src/plans/quasi_newton_plan.jl

```diff
@@ -478,14 +478,15 @@ function (d::QuasiNewtonLimitedMemoryDirectionUpdate{InverseBFGS})(mp, st)
     for i in m:-1:1
         d.ρ[i] = 1 / inner(M, p, d.memory_s[i], d.memory_y[i]) # 1 sk 2 yk
         d.ξ[i] = inner(M, p, d.memory_s[i], r) * d.ρ[i]
-        r .= r .- d.ξ[i] .* d.memory_y[i]
+        r .-= d.ξ[i] .* d.memory_y[i]
     end
     r .= 1 / (d.ρ[m] * norm(M, p, last(d.memory_y))^2) .* r
     for i in 1:m
-        r .= r .+ (d.ξ[i] - d.ρ[i] * inner(M, p, d.memory_y[i], r)) .* d.memory_s[i]
+        r .+= (d.ξ[i] - d.ρ[i] * inner(M, p, d.memory_y[i], r)) .* d.memory_s[i]
     end
-    d.project && project!(M, r, p, r)
-    return -r
+    d.project && embed_project!(M, r, p, r)
```

Comment:
I am also not sure here. Besides that we currently have 2 different forms of specifying this (I think in TCG we store a function here we have a bool) I am not sure why this switch was necessary. In the old code was no embed. Can you help me understand why that was necessary? This line is only meant to introduce stability of the algorithm.

Reply:
Simply, `project` is not idempotent. For example on `Rotations` it isn't. `embed_project!` is idempotent. In generic code you shouldn't call `project` before  you actually `embed` a point or tangent vector.

Reply:
Hm, then most of my code is probably wrong :(

Reply:
:(

Reply:
So just historically I up to the discussion we had and this extra function we use – have used `project!` in several places for stability.